gettsim/taxes/eink_st.py

```diff
@@ -53,37 +56,8 @@ def st_tarif(x, params):
 
     args:
         x (float): taxable income
-        tb (dict): tax-benefit parameters specific to year and reform
+        params (dict): tax-benefit parameters specific to year and reform
     """
 
-    if params["jahr"] < 2002:
-        raise ValueError("Income Tax Pre 2002 not yet modelled!")
-    else:
-        eink_steuer = 0.0
-        if params["G"] < x <= params["M"]:
-            eink_steuer = (
-                ((params["t_m"] - params["t_e"]) / (2 * (params["M"] - params["G"])))
-                * (x - params["G"])
-                + params["t_e"]
-            ) * (x - params["G"])
-        elif params["M"] < x <= params["S"]:
-            eink_steuer = (
-                ((params["t_s"] - params["t_m"]) / (2 * (params["S"] - params["M"])))
-                * (x - params["M"])
-                + params["t_m"]
-            ) * (x - params["M"]) + (params["M"] - params["G"]) * (
-                (params["t_m"] + params["t_e"]) / 2
-            )
-        elif x > params["S"]:
-            eink_steuer = (
-                params["t_s"] * x
-                - params["t_s"] * params["S"]
-                + ((params["t_s"] + params["t_m"]) / 2) * (params["S"] - params["M"])
-                + ((params["t_m"] + params["t_e"]) / 2) * (params["M"] - params["G"])
-            )
-        if x > params["R"]:
-            eink_steuer = eink_steuer + (params["t_r"] - params["t_s"]) * (
-                x - params["R"]
-            )
-        assert eink_steuer >= 0
-    return eink_steuer
+    eink_st = piecewise_polynominal(x, **params["eink_st_tarif"])
```

Comment:
What to the two stars indicate?

Reply:
The two stars are for mapping keywords to values in the function. See https://www.geeksforgeeks.org/args-kwargs-python/ for details!

Reply:
Have not checked the example closely, but in general, try to be explicit and write them out. Makes it much easier to understand the code while reading, let alone debugging.

Reply:
Replaced with explicit statement